tests/unit/test_gitlab_http_methods.py

```diff
@@ -38,6 +38,100 @@ def resp_cont(url, request):
             gl.http_request("get", "/not_there")
 
 
+def test_http_request_302_get(gl):
+    """Test to show that a redirect of a GET will not cause an error"""
+
+    @urlmatch(
+        scheme="http", netloc="localhost", path="/api/v4/user/status", method="get"
+    )
+    def resp_cont(url, request):
+        req = requests.Request(
+            "GET",
+            url="http://localhost//api/v4/user/status",
+        )
+        prepped = req.prepare()
+
+        history = [
+            response(
+                status_code=302,
+                content=(
+                    b"<html>\r\n<head><title>302 Found</title></head>\r\n<body bgcolor"
+                    b'="white">\r\n<center><h1>302 Found</h1></center>\r\n<hr><center>'
+                    b"nginx</center>\r\n</body>\r\n</html>\r\n"
+                ),
+                headers={"Location": "https://example.com/api/v4/user/status"},
+                reason="Moved Temporarily",
+                request=prepped,
+            )
+        ]
+
+        content = (
+            b'{"emoji":"thumbsup","message":"Hi redirect!","availability":"not_set",'
+            b'"message_html":"Hi redirect!","clear_status_at":null}'
+        )
+        resp_obj = response(
+            status_code=200,
+            content=content,
```

Comment:
couldn't this be just formatted as a Python object (dict) and provide it as a `json` argument?

Reply:
I'm not sure. I changed things around and added some comments which hopefully make things clearer.